### ISSUE TITLE
(Feature Request): Badge if a content creator gave access to their private dislikes

### ISSUE THREAD

**Issue by DerpedCrusader** [Anarios/return-youtube-dislike#130]
### Extension or Userscript?

Both

### Request or suggest a new feature!

A badge to the right of the verified checkmark if that specific content creator has allowed the API to scrape their dislike information, so you can be certain that their dislikes are "verified"

### Ways to implement this!

I was hoping for maybe a little badge next to their name, I made a small mock up:

![Screenshot 2021-11-30 200502](https://user-images.githubusercontent.com/13917665/144165348-9bd9f7c0-e55e-481e-a46f-87957997a343.png)

for the "can you work on this" i could make a better dislike design, but other than that, I have 0 coding experience

### Can you work on this?

- [ ] Yes
- [X] No

**Comment by hisamafahri** [comment]
I really love this idea. 

I have made some badge designs in SVG for both dark and light theme. Feel free to redesign, or do anything since it's just based on my preferences. These images below are just preview using DevTools.

- *Rounded style - Dark Theme*
![SCR-20211201-iu9](https://user-images.githubusercontent.com/65691613/144184984-6de771d4-7f38-47eb-a94d-6a49bc5a557b.png)

- *Rounded style - Light Theme*
![SCR-20211201-ix5](https://user-images.githubusercontent.com/65691613/144185049-ac66608a-eeee-4181-a4ee-23280322a8b8.png)

- *Hexagon style - Dark Theme*
![SCR-20211201-ixp](https://user-images.githubusercontent.com/65691613/144185145-8a5f44d4-cb35-4614-a18b-fd13aaaa6662.png)

- *Hexagon style - Light Theme*
![SCR-20211201-iwo](https://user-images.githubusercontent.com/65691613/144185190-6416f995-4cfc-4739-bace-5e93a0dbfc0c.png)



You can download the SVG File:
[20x20-dislike-badge.zip](https://github.com/Anarios/return-youtube-dislike/files/7631434/20x20-dislike-badge.zip)

Or visit the Figma project: [https://www.figma.com/file/2f0kBjLlt0V5E9NfxGWSk9/Dislike-Badge?node-id=0%3A1](https://www.figma.com/file/2f0kBjLlt0V5E9NfxGWSk9/Dislike-Badge?node-id=0%3A1)

**Comment by mattphelps-8451** [comment]
@hisamafahri 
What about "Public Dislike Contributor" instead?

**Comment by hisamafahri** [comment]
@mattphelps-8451 That's a good idea

**Comment by DerpedCrusader** [comment]
the hexagon looks nice, and contrasts with the circle check mark if their verified by youtube

**Comment by sy-b** [comment]
This is similar to #108

**Comment by sy-b** [comment]
### My thoughts

- The batch must reflect "positivity", as it is there to create trust among the viewers.
- The design should be mote colorful & cheerful.
- Shouldn't be confused with ✅ badge by YouTube
- Placing the badge beside the Channel Name, might confuse people that its by YouTube, since "verification" (✅) bagges are given only by YT & placed there. (i.e. lets not (even slightly) offend Google by placing it near _their_ badge)

**Comment by Greedquest** [comment]
I'd say just use the icon of this extension, to show they contribute data to this extension

**Comment by laughingsheep** [comment]
May I suggest something like that? It is more neutral/positive than the original Design. (Inspired/Copied from @hisamafahri idea) 
![viewyt](https://user-images.githubusercontent.com/61907641/144249047-7b161693-38a6-4b84-9298-afb814afd1c4.png)
![comboyt](https://user-images.githubusercontent.com/61907641/144249051-5f076a4f-1512-4a2b-877c-d04dcda86d01.png)

**Comment by laughingsheep** [comment]
Or how about a using the place above or below the line and displaying the information the
![Info](https://user-images.githubusercontent.com/61907641/144252691-7a7d2cde-9792-4c94-afe8-69c40534f4c7.png)
![Implemented](https://user-images.githubusercontent.com/61907641/144252695-463c39cb-609b-4bd0-bfec-ae9c516b5d37.png)
re

**Comment by laughingsheep** [comment]
Or using the Space above or below the line and displaying the information there.
![Example](https://user-images.githubusercontent.com/61907641/144253109-8fe94d0b-07cc-4a18-9643-752a3ca8d679.png)

**Comment by irlcatgirl** [comment]
I mentioned this on the Discord yesterday. Thanks for getting an issue created for this. Some feedback the badge is a bit small, and this might be an accessibility issue for visually impaired users. Also do we have a pull request to look at yet, would be interested in taking on making sure the aria-tags are setup correctly.

**Comment by rdev** [comment]
Thoughts? 
Adding a badge to the dislike button itself and using Youtube's tooltip style feels the most organic
Making it green gets attention and lets people know immediately that dislike data is legit and creator chose to share it

## Option 1

![Artboard](https://user-images.githubusercontent.com/8418866/144291687-b407408c-7f48-4d42-b1eb-674eed2d912f.png)


## Option 2
![Artboard](https://user-images.githubusercontent.com/8418866/144291291-1e710ab5-75af-4dee-88d9-da5a298938b1.png)

**Comment by irlcatgirl** [comment]
I prefer option 2. Option 1 is a little off putting in terms of placement.

Green is easier to see than gray, could you look at making it red like the subscribe option? I’m curious what it’d look like.

**Comment by rdev** [comment]
Feels a bit like too much IMO 

![Artboard](https://user-images.githubusercontent.com/8418866/144292357-26781fd8-5a97-4fd7-995d-3dc9e36f03be.png)

**Comment by irlcatgirl** [comment]
Could you change the checkmark from black to light gray or white?

**Comment by rdev** [comment]
The check is transparent, so it'll be white in light mode and drak gray in dark mode

Here's what both colors look like with forced white fill

## Red + White

![Artboard](https://user-images.githubusercontent.com/8418866/144293130-8b7b6527-6163-47ca-871d-64d29617cfa1.png)

## Green + White

![Artboard](https://user-images.githubusercontent.com/8418866/144293158-838f8468-99bb-4520-bb50-2de64f190b6d.png)

**Comment by bonn2** [comment]
I like the inline view, maybe we make this an option in the add-on to pick the image you want. I would personally like to see a white outline-style one that matches the other icons in the bar, but the green+white is also very nice.

**Comment by rdev** [comment]
@bonn2 I like the outline idea

![Artboard](https://user-images.githubusercontent.com/8418866/144295029-3fcf2840-bb54-4982-9d20-a64c09065c73.png)

**Comment by irlcatgirl** [comment]
I personally prefer Red+White. It looks better to me and fits in with the YouTube Red color. Green seems a bit out of place personally.

**Comment by rdev** [comment]
@irlcatgirl I agree with @sy-b's point that it should have a positive reinforcement effect. Youtube red color doesn't have that effect while green does 🤔

**Comment by sy-b** [comment]
How would a normal user get to know its meaning.
Wouldn't then get it confused for some kind of official verification from YouTube

Suggestion: use words also

**Comment by sy-b** [comment]
& what if the data is gathered by other sources?
#108 takes this into account. 

If the aim of this discussion is expanded to include other sources, then I'll close #108

**Comment by bonn2** [comment]
> How would a normal user get to know its meaning. Wouldn't then get it confused for some kind of official verification from YouTube
> 
> Suggestion: use words also

Maybe instead of a checkmark it could be a little dislike icon in the symbol

**Comment by irlcatgirl** [comment]
I like the idea of adding a settings page to pick the icon. Perhaps include an option for custom CSS/SVG for the icon.

**Comment by Anarios** [comment]
Green + white looks best to me. Red + white is good as well, but green is better to convey "positive" message.

**Comment by irlcatgirl** [comment]
We should have a ? Icon perhaps colored yellow or orange to show we are estimating it vs know for a fact. I agree a tool tip would help here as well.

**Comment by sy-b** [comment]
Should that badge be adaptable to theme?

**Comment by irlcatgirl** [comment]
I think that would be a good idea

**Comment by Cederick** [comment]
> Green + white looks best to me. Red + white is good as well, but green is better to convey a "positive" message.
Personally, I'd like to have some indicator that does not future distract from the actual content by being bright and flashy

**Comment by irlcatgirl** [comment]
> > Green + white looks best to me. Red + white is good as well, but green is better to convey "positive" message.
> 
> Personally, I'd like to have some indicator that does not future distract from the actual content by being bright and flashy

I'm indecisive here, I would say that the rest of the community should chime in for the default, I would only ask its left as a configurable flag. Perhaps use Green by default but allow people to swap to the YouTube Red one. I think user choice is important here but am far from a UX expert, my idea may be bad, so I'll ask the rest of the community to vote / discuss about this.

**Comment by UnderscoreX5** [comment]
> @bonn2 I like the outline idea
> 
> ![Artboard](https://user-images.githubusercontent.com/8418866/144295029-3fcf2840-bb54-4982-9d20-a64c09065c73.png)
> 

Looks great

**Comment by hrichiksite** [comment]
This is so great, it would integrate with my project as well.

https://github.com/Anarios/return-youtube-dislike/issues/401
